heartbeat/gcp-vpc-move-vip.in

```diff
@@ -0,0 +1,315 @@
+
+  # Loops through all hosts & remove the alias IP from the host that has it
+  hostlist = os.environ.get('OCF_RESKEY_hostlist', '')
+  if hostlist:
+    hostlist.replace(THIS_VM, '').split()
+  else:
+    hostlist = get_instances_list(project, THIS_VM)
+  for host in hostlist:
+    host_zone = get_zone(project, host)
+    host_alias = get_alias(project, host_zone, host)
+    if alias == host_alias:
+      logging.info(
+          '%s is attached to %s - Removing all alias IP addresses from %s' %
+          (alias, host, host))
+      set_alias(project, host_zone, host, '')
+      break
+
+  # add alias IP to localhost
+  set_alias(
+      project, my_zone, THIS_VM, alias,
+      os.environ.get('OCF_RESKEY_alias_range_name'))
+
+  # Check the IP has been added
+  my_alias = get_localhost_alias()
+  if alias == my_alias:
+    logging.info('Finished adding %s to %s' % (alias, THIS_VM))
+  elif my_alias:
+    logging.error(
+        'Failed to add IP. %s has an IP attached but it isn\'t %s' %
+        (THIS_VM, alias))
+    sys.exit(OCF_ERR_GENERIC)
+  else:
+    logging.error('Failed to add IP address %s to %s' % (alias, THIS_VM))
+    sys.exit(OCF_ERR_GENERIC)
+
+
+def gcp_alias_stop(alias):
+  my_alias = get_localhost_alias()
+  my_zone = get_metadata('instance/zone').split('/')[-1]
+  project = get_metadata('project/project-id')
+
+  if my_alias == alias:
+    logging.info('Removing %s from %s' % (my_alias, THIS_VM))
+    set_alias(project, my_zone, THIS_VM, '')
+
+
+def gcp_alias_status(alias):
+  my_alias = get_localhost_alias()
+  if alias == my_alias:
+    logging.info('%s has the correct IP address attached' % THIS_VM)
+  else:
+    sys.exit(OCF_NOT_RUNNING)
+
+
+def configure():
+  global CONN
+  global THIS_VM
+
+  # Populate global vars
+  CONN = googleapiclient.discovery.build('compute', 'v1')
+  THIS_VM = get_metadata('instance/name')
+
+  # Prepare logging
+  logging.basicConfig(
+      format='gcp:alias - %(levelname)s - %(message)s', level=logging.INFO)
+  logging.getLogger('googleapiclient').setLevel(logging.WARN)
+  logging_env = os.environ.get('OCF_RESKEY_stackdriver_logging')
+  if logging_env:
+    logging_env = logging_env.lower()
+    if any(x in logging_env for x in ['yes', 'true', 'enabled']):
+      try:
+        import google.cloud.logging.handlers
+        client = google.cloud.logging.Client()
+        handler = google.cloud.logging.handlers.CloudLoggingHandler(
+            client, name=THIS_VM)
+        handler.setLevel(logging.INFO)
+        formatter = logging.Formatter('gcp:alias "%(message)s"')
+        handler.setFormatter(formatter)
+        root_logger = logging.getLogger()
+        root_logger.addHandler(handler)
+      except ImportError:
+        logging.error('Couldn\'t import google.cloud.logging, '
+            'disabling Stackdriver-logging support')
+
+
+def main():
+  if 'meta-data' in sys.argv[1]:
+    print(METADATA)
+    return
+
+  alias = os.environ.get('OCF_RESKEY_alias_ip')
+  if not alias:
```

Comment:
We still need it as a validate-action (both in metadata and sys.argv[1] below:
https://github.com/ClusterLabs/resource-agents/blob/master/heartbeat/aws-vpc-move-ip#L132

Reply:
I see, ok, I'll update the PR

Reply:
@oalbrigt could you please check my latest commit? I am not sure if I got this right